src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -34,15 +35,36 @@ def create_from_stripped_path(cls, path: PurePath) -> "PythonModule":
 
 
 @dataclass(frozen=True)
-class FirstPartyModuleToAddressMapping:
+class PythonFirstPartyModuleMappingPlugin:
+    """A mapping of module names to owning addresses that a plugin adds for Python import dependency
+    inference."""
+
+    mapping: FrozenDict[str, Address]
+
+
+@union
+class PythonFirstPartyModuleMappingPluginRequest:
+    """A request for a plugin to create a mapping of module names to owning addresses to be added
+    for Python import dependency inference.
+
+    These modules will be combined with the built-in first-party module mapping. Any conflicting
+    modules will be removed due to ambiguity.
+
+    The addresses should all be file addresses, rather than BUILD addresses.
+    """
+
+
+@dataclass(frozen=True)
+class PythonFirstPartyModuleMapping:
     """A mapping of module names to owning addresses.
 
     All mapped addresses will be file addresses, aka generated subtargets. That is, each target
     will own no more than one single source file.
 
-    If there are >1 original owning targets that refer to the same module—such as `//:a` and `//:b` both owning module
-    `foo`—then we will not add any of the targets to the mapping because there is ambiguity. (We make an exception if
-    one target is an implementation (.py file) and the other is a type stub (.pyi file).
+    If there are >1 original owning targets that refer to the same module—such as `//:a` and `//:b`
+    both owning module `foo`—then we will not add any of the targets to the mapping because there
+    is ambiguity. (We make an exception if one target is an implementation (.py file) and the other
+    is a type stub (.pyi file).
```

Comment:
This file had a couple of comments way over 100 character limit. I reformatted them but didn't make changes.